Lesson33/Gokhan/src/App.js

```diff
@@ -30,23 +31,26 @@ export const App = () => {
 	};
 
 	const handleSignOut = () => {
+		dispatch({
+			type: 'LOGOUT',
+		});
 		setUser(loggedOutUser);
 		setShowAccountPage(false);
 		setShowListPage(false);
 	};
 
-	const handleUserNameChange = (newUserName) => {
-		setUser((prevUser) => ({ ...prevUser, name: newUserName }));
-	};
+	// const handleUserNameChange = (newUserName) => {
+	// 	setUser((prevUser) => ({ ...prevUser, name: newUserName }));
+	// };
 
 	const complateTodos = todos.filter(todo => todo.completed);
 
 	return (
-		<>
+		<UserProvider>
```

Comment:
Move UserProvider to `src/index.js` file, this way you'll be able to use user data in App.js file.